site/src/pages/TemplateVersionEditorPage/ProvisionerTagsPopover.tsx

```diff
@@ -0,0 +1,130 @@
+
+import { Stack } from 'components/Stack/Stack';
+import { TopbarButton } from 'components/FullPageLayout/Topbar';
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "components/Popover/Popover";
+import { ProvisionerTag } from 'pages/HealthPage/ProvisionerDaemonsPage';
+import { type FC} from 'react';
+import useTheme from '@mui/system/useTheme';
+import { useFormik } from 'formik';
+import * as Yup from "yup";
+import { getFormHelpers, onChangeTrimmed } from 'utils/formUtils';
+import { FormFields, FormSection, VerticalForm } from 'components/Form/Form';
+import TextField from '@mui/material/TextField';
+import Button from '@mui/material/Button';
+import ExpandMoreOutlined from '@mui/icons-material/ExpandMoreOutlined';
+import AddIcon from '@mui/icons-material/Add';
+
+const initialValues = {
+  key: "",
+  value: "",
+};
+
+const validationSchema = Yup.object({
+  key: Yup.string().required("Required").notOneOf(["owner"], "Cannot override owner tag"),
+  value: Yup.string().required("Required").when("key", ([key], schema) => {
+    if (key === "scope") {
+      return schema.oneOf(["organization", "scope"], "Scope value must be 'organization' or 'user'");
+    }
+
+    return schema;
+  })
+});
+
+interface ProviderTagsPopoverProps {
+  tags: Record <string, string>;
+  onSubmit: (values: typeof initialValues) => void;
+  onDelete: (key: string) => void;
+}
+
+export const ProviderTagsPopover: FC<ProviderTagsPopoverProps> = ({ tags, onSubmit, onDelete }) => {
+  const theme = useTheme();
+
+  const form = useFormik({
+    initialValues,
+    validationSchema,
+    onSubmit: (values) => {
+      onSubmit(values);
+      form.resetForm();
+    },
+  });
+  const getFieldHelpers = getFormHelpers(form);
+
+  return (
+    <Popover isDefaultOpen={false}>
+    <PopoverTrigger>
+      <TopbarButton
+        data-testid="build-parameters-button"
```

Comment:
This is probably not used here for tests. If it is, it should have a better name.